### PR TITLE
Use C++11 std::thread instead of pthread.

### PR DESCRIPTION
**Pull request by mgautierfr**
This is necessary as libzim now does the same and we don't install wrapper pthread library on windows CI.

### CODE REVIEW

File: src/subprocess_unix.cpp [hunk 1]
```diff
@@ -11,47 +11,30 @@
...
-  m_mutex(PTHREAD_MUTEX_INITIALIZER),
-  m_waitingThread()
+  m_running(false)
 {
 }
 
 UnixImpl::~UnixImpl()
 {
   kill();
 // Android has no pthread_cancel :(
```
Comment (veloman-yunkan) [inline src/subprocess_unix.cpp:21]: This comment doesn't make sense anymore.

File: src/subprocess_unix.cpp [hunk 2]
```diff
@@ -11,47 +11,30 @@
...
-  sigaction(SIGUSR1, &actions, NULL);
-#endif
-
   UnixImpl* self = static_cast<UnixImpl*>(_self);
-  waitpid(self->m_pid, NULL, 0);
+  while (true) {
+    if (!waitpid(self->m_pid, NULL, WNOHANG)) {
+      break;
+    }
+    if (self->m_shouldQuit) {
```
Comment (veloman-yunkan) [inline src/subprocess_unix.cpp:32]: I couldn't find where `m_shouldQuit` is set

File: src/subprocess_unix.cpp [hunk 3]
```diff
@@ -11,47 +11,30 @@
 
 UnixImpl::UnixImpl():
   m_pid(0),
-  m_running(false),
-  m_mutex(PTHREAD_MUTEX_INITIALIZER),
-  m_waitingThread()
+  m_running(false)
 {
 }
```
Comment (veloman-yunkan) [inline src/subprocess_unix.cpp:16]: `m_shouldQuit` is left uninitialized

**Review by veloman-yunkan — CHANGES_REQUESTED** [review]

File: src/subprocess_unix.cpp (hunk 1 above)
Comment (mgautierfr) [reply src/subprocess_unix.cpp]: Removed.

File: src/subprocess_unix.cpp (hunk 2 above)
Comment (mgautierfr) [reply src/subprocess_unix.cpp:33]: Added in the destructor.

File: src/subprocess_unix.cpp (hunk 3 above)
Comment (mgautierfr) [reply src/subprocess_unix.cpp]: Done

**Review by veloman-yunkan — APPROVED** [review]

**Pull request merged**